cg/models/demultiplex/run_parameters.py

```diff
@@ -1,31 +1,116 @@
+"""Module for modeling run parameters file parsing."""
 import logging
 from pathlib import Path
-from typing import Optional
+from typing import Optional, Dict
 from xml.etree import ElementTree
 
 from cg.constants.demultiplexing import UNKNOWN_REAGENT_KIT_VERSION
-from cg.exc import FlowCellError
+from cg.constants.sequencing import Sequencers
+from cg.exc import RunParametersError
 
 LOG = logging.getLogger(__name__)
 
 
 class RunParameters:
-    """Class to handle the run parameters from a sequencing run."""
+    """Base class with basic functions to handle the run parameters from a sequencing run."""
 
     def __init__(self, run_parameters_path: Path):
         self.path: Path = run_parameters_path
         with open(run_parameters_path, "rt") as in_file:
             self.tree: ElementTree = ElementTree.parse(in_file)
+        self.validate_instrument()
 
     @property
     def index_length(self) -> int:
         """Return the length of the indexes if they are equal, raise an error otherwise."""
         index_one_length: int = self.get_index1_cycles()
         index_two_length: int = self.get_index2_cycles()
         if index_one_length != index_two_length:
-            raise FlowCellError("Index lengths are not the same!")
+            raise RunParametersError("Index lengths are not the same!")
         return index_one_length
 
+    @staticmethod
+    def node_not_found(node: Optional[ElementTree.Element], name: str) -> None:
+        """Raise exception if the given node is not found."""
+        if node is None:
+            message = f"Could not determine {name}"
+            LOG.warning(message)
+            raise RunParametersError(message)
+
+    def validate_instrument(self) -> None:
+        """Raise an error if the parent class was instantiated."""
+        raise NotImplementedError(
+            "Parent class instantiated. Instantiate instead RunParametersNovaSeq6000 or RunParametersNovaSeqX"
+        )
+
+    @property
+    def requires_dummy_samples(self) -> Optional[bool]:
+        """Return true if the flow cell requires the addition of dummy samples."""
+        raise NotImplementedError("Impossible to know dummy sample requirements of parent class")
+
+    @property
+    def control_software_version(self) -> Optional[str]:
+        """Return the control software version if existent."""
+        raise NotImplementedError(
+            "Impossible to retrieve control software version from parent class"
+        )
+
+    @property
+    def reagent_kit_version(self) -> Optional[str]:
+        """Return the reagent kit version if existent."""
+        raise NotImplementedError("Impossible to retrieve reagent kit version from parent class")
+
+    @property
+    def sequencer(self) -> Optional[str]:
+        """Return the sequencer associated with the current run parameters."""
+        raise NotImplementedError("Impossible to retrieve sequencer from parent class")
+
+    def get_index1_cycles(self) -> int:
```

Comment:
Not a fan of mixing numbers in method names, but I guess it is a matter of taste. Should it be separated with an underscore? Or maybe say `get_first_index_cycles`, `get_index_1_cycles` or `get_index_one_cycles` instead? Same for the ones below, something to consider

Reply:
`index1` and `index2` are the names of the columns of the v2 sample sheet. I am not 100% sure, but I think that this is something that comes from Illumina. In either case, changing it in cg would be a big job that won't give us a lot of benefit, I think.